std/container/slist.d

```diff
@@ -135,6 +135,48 @@ struct SList(T)
         return n;
     }
 
+    private static auto createNodeChain(Stuff)(Stuff stuff)
+    if (isImplicitlyConvertible!(Stuff, T))
+    {
+        import std.range : only;
+        return createNodeChain(only(stuff));
+    }
+
+    private static auto createNodeChain(Stuff)(Stuff stuff)
+    if (isInputRange!Stuff && isImplicitlyConvertible!(ElementType!Stuff, T))
+    {
+        static struct Chain
+        {
+            Node* first;
+            Node* last;
+            size_t length;
+        }
+
+        Chain ch;
+
+        foreach (item; stuff)
+        {
+            auto newNode = new Node(null, item);
+            (ch.first ? ch.last._next : ch.first) = newNode;
+            ch.last = newNode;
+            ++ch.length;
```

Comment:
AFAICT you don't need that. Just check whether `first !is null`? (probably a good idea to add an `empty` method for this to Chain to encapsulate things)

Reply:
the `length` field is there just so `insertFront` and `insertAfter` can return the number of elements inserted, as they are supposed to do.